### PR TITLE
F#src 5238 better pr testing

### PR DESCRIPTION
**Pull request by ToivoS**

### CODE REVIEW

**Review by biofotis — CHANGES_REQUESTED** [review]
It might not be a good time now but the run-ansible.sh is getting as big as the first oneliner that we tried to get away from.

File: ansible/roles/products/common/docker-container/tasks/reinstall.yml [hunk 1]
```diff
@@ -24,6 +24,33 @@
...
+    template:
+      src: ../../../common/resources/templates/scripts/pull-pr-branches-and-catkin-make.j2
+      dest: "{{ shadow_hand_launcher_folder }}/pull_pr_branches_and_catkin_make.sh"
+      mode: '755'
+    vars:
+      pr_branches_value: "{{ pr_branches }}"
+
+  - name: Copying the pull pull_pr_branches_and_catkin_make bash script into docker container
+    changed_when: false
+    shell: "docker cp {{ shadow_hand_launcher_folder }}/pull_pr_branches_and_catkin_make.sh {{ container_name }}:/home/user/pull_pr_branches_and_catkin_make.sh"
```
[the remark this answers is not in the document]
Comment (ToivoS) [reply ansible/roles/products/common/docker-container/tasks/reinstall.yml]: yes, it's fixed in the latest commit

**Comment by ToivoS** [comment]
> It might not be a good time now but the run-ansible.sh is getting as big as the first oneliner that we tried to get away from.

**Comment by ToivoS** [comment]
> It might not be a good time now but the run-ansible.sh is getting as big as the first oneliner that we tried to get away from.

run-ansible.sh is getting big because:
1) we do user-input validation (first oneliner didn't do that)
2) allows user-entered arguments which contain spaces inside quotes "val1 val2 val3" (first oneliner didn't do that)
3) checks if the user has got an SSH key which is registered with Github and asks user to create it while waiting for confirmation and then checking again (first oneliner didn't do that)
4) it's a good idea to do this input argument validation, handling spaces and SSH checks in run-ansible.sh because it avoids waiting a long time while ansible runs and goes to the correct section in the playbook (after installing docker, etc.) to check for those variables (and we would have to perform this check in multiple playbooks and multiple sections of the same playbook)

**Review by biofotis — APPROVED** [review]

**Review by biofotis — APPROVED** [review]

**Pull request merged**